ospf6d/ospf6_lsa.c

```diff
@@ -551,27 +551,49 @@ void ospf6_lsa_show_dump(struct vty *vty, struct ospf6_lsa *lsa,
 {
 	uint8_t *start, *end, *current;
 	char byte[4];
+	char *header_str;
+	char adv_router[64];
+	char id[64];
+	json_object *json;
 
 	start = (uint8_t *)lsa->header;
 	end = (uint8_t *)lsa->header + ntohs(lsa->header->length);
 
-	if (use_json)
-		return;
+	if (use_json) {
+		json = json_object_new_object();
+		header_str = (char *)(malloc((2 * (end - start)) + 1));
```

Comment:
any particular reason we aren't using XMALLOC?

Reply:
Fixed; switched to `XMALLOC` with `MTYPE_TMP`.